corehq/apps/cloudcare/views.py

```diff
@@ -111,6 +111,9 @@ def fetch_app(self, domain, app_id):
         return _fetch_build(domain, self.request.couch_user.username, app_id)
 
     def get_web_apps_available_to_user(self, domain, user):
+        if user['doc_type'] == 'WebUser' and not user.can_access_web_apps(domain):
```

Comment:
I wonder if this logic belongs better in `user_can_access_app`

Reply:
I'd keep them separate - `user_can_access_app` is about to become legacy functionality (https://github.com/dimagi/commcare-hq/pull/34421).